src/test/java/xml/XmlReaderTest.java

```diff
@@ -27,96 +30,92 @@
 	@Before
 	public void setUp() {
 		assertNotNull("Test file not found", getClass().getResource("/user_save.xml"));
-		try {
-			file = new File(getClass().getResource("/user_save.xml").getFile());
-			reader = new XmlReader(file) ;
-		} catch (ParserConfigurationException | SAXException | IOException e) {
-			e.printStackTrace();
-		}
+		file = new File(getClass().getResource("/user_save.xml").getFile());
+		reader = new XmlReader();
 	}
 
 	@Test
 	public void testReadStreamedXMLversion() {
-		assertEquals("1.0", reader.getDocument().getXmlVersion());			
+		try {
+			assertEquals("1.0", reader.read(file).getXmlVersion());
+		} catch (ParserConfigurationException | SAXException | IOException e) {
+			e.printStackTrace();
```

Comment:
Meestal doe je bij tests gewoon throws Exception en laat je JUnit het als een fail weergeven als het fout gaat, dat is veel mooier dan dit, want dit resulteert niet in een gefaalde test.